tcl/response.tcl

```diff
@@ -1,18 +1,30 @@
 namespace eval qc::response {
 
-    namespace export status record message action extend
+    namespace export status record message action extend data
     namespace ensemble create -unknown {
         response_subcommand_map
     }
 
+    proc data {tson} {
+        #| Sets the JSON response object data property using tson
+        global data
+        dict set data data $tson
+    }
+
     proc extend {name args} {
         #| Extends the JSON response with an object named $name with properties defined in $args.
         if { [llength $args] % 2 != 0 } {
             return -code error "Usage: qc::response extend name key value ?key value ...?"
         }
         global data
-        dict for {key value} $args {
-            dict set data $name $key $value
+        if { $name eq "data" } {
```

Comment:
I think if we wanted to set the response data using tson we would just use `qc::response data $data` .
If the developer used `qc::response extend data value` then I think it is reasonable that we keep the existing behavior that does not support tson structures.

I'd imagine that we will most probably use `qc::response data ...` instead of `qc::response extend ...` going forward.

Any other situations that you think this functionality would be required.